src/Coinbase/Exchange/Socket.hs

```diff
@@ -19,16 +18,10 @@ import           Coinbase.Exchange.Types.Socket
 
 
 -------------------------------------------------------------------------------
-subscribe :: ApiType -> [ProductId] -> WS.ClientApp a -> IO a
-subscribe atype pids app = withSocketsDo $
+subscribe :: ApiType -> [ProductId] -> [Channel] -> WS.ClientApp a -> IO a
+subscribe atype pids chans app = withSocketsDo $
         runSecureClient location 443 "/" $ \conn -> do
-            WS.sendTextData conn $ encode (Subscribe pids)
+            WS.sendTextData conn $ encode (Subscribe pids chans)
             app conn
     where location = case atype of Sandbox -> sandboxSocket
                                    Live    -> liveSocket
-
-
--------------------------------------------------------------------------------
--- | Enable/disable heartbeat anytime you want in your 'ClientApp'
```

Comment:
Why was this removed? Does it not work anymore?

Reply:
I'm not sure what the API used to look like, but in the current API heartbeats are implemented as a channel, which you can subscribe to in the same was as other channels. Ordinarily, you would do a single `subscribe` with the list of channels you want to listen on, including `Heartbeat` in that list if you want heartbeats. We can re-implement `setHeartbeat` using `subscribe` for backwards compatibility if you want, but otherwise I don't think heartbeats deserve to be special-cased.

Reply:
Makes sense